src/mctpd.c

```diff
@@ -288,6 +289,53 @@ static peer * find_peer_by_addr(ctx *ctx, mctp_eid_t eid, int net)
 	return NULL;
 }
 
+bool find_local_eids_count_by_net(ctx *ctx, uint32_t net, size_t *num)
+{
+	net_det *n = lookup_net(ctx, net);
+	peer *peer;
+	*num = 0;
+
+	if (!n)
+		return false;
+
+	for (size_t t = 0; t < 256; t++) {
+		if (n->peeridx[t] < 0)
+			continue;
+
+		peer = &ctx->peers[n->peeridx[t]];
+		if (peer && (peer->state == LOCAL))
+		{
+			*num = *num + 1;
+		}
+	}
+
+	return true;
+}
+
+mctp_eid_t * find_local_eids_by_net(ctx *ctx, uint32_t net,
+                                    const size_t local_eid_cnt)
+{
+	mctp_eid_t *ret_eids = (mctp_eid_t *)malloc(local_eid_cnt);
+	size_t local_count = 0;
+	net_det *n = lookup_net(ctx, net);
+	peer *peer;
+
+	if (!n)
+		return NULL;
+
+	for (size_t t = 0; t < 256; t++) {
+		if (n->peeridx[t] < 0)
+			continue;
+
+		peer = &ctx->peers[n->peeridx[t]];
+		if (peer && (peer->state == LOCAL))
+			ret_eids[local_count++] = t;
+	}
+
+	return ret_eids;
+}
```

Comment:
You have essentially the same logic repeated here. Just return the count and array from the one function.